pandas/io/stata.py

```diff
@@ -792,19 +802,35 @@ class StataMissingValue:
         "float64": struct.unpack("<d", float64_base)[0],
     }
 
-    def __init__(self, value):
+    def __init__(self, value: Union[int, float]) -> None:
```

Comment:
in pandas we choose not to annotate the return type of \_\_init__ and int is a subtype of float, see
https://mypy.readthedocs.io/en/latest/duck_type_compatibility.html#duck-type-compatibility
```suggestion
    def __init__(self, value: float):
```

Reply:
I've converted these all to float, even though I like Union[int, float] a bit better in these functions since the int and float parts are distinct (e.g., looking up sentinal values).